conda_smithy/configure_feedstock.py

```diff
@@ -595,9 +598,96 @@ def _load_forge_config(forge_dir, variant_config_files):
+    from conda_build.conda_interface import VersionOrder, MatchSpec
+    available_versions = [pkg.version for pkg in resolve.get_pkgs(MatchSpec(name))]
+    available_versions = sorted(available_versions, key=VersionOrder)
+    most_recent_version = available_versions[-1]
+    if installed_version is None:
+        msg = "{} is not installed in root env.".format(name)
+    elif VersionOrder(installed_version) < VersionOrder(most_recent_version):
+        msg = "{} version in root env is out-of-date.".format(name)
+    else:
+        return
+    if error_on_warn:
+        raise RuntimeError("{} Exiting.".format(msg))
+    else:
+        print(msg)
+
+
+def commit_changes(forge_file_directory, commit, cs_ver, cfp_ver):
+    if cfp_ver:
+        msg = 'Re-rendered with conda-smithy {} and pinning {}'.format(cs_ver, cfp_ver)
+    else:
+        msg = 'Re-rendered with conda-smithy {}'.format(cs_ver)
+    print(msg)
+
+    is_git_repo = os.path.exists(os.path.join(forge_file_directory, ".git"))
+    if is_git_repo:
+        has_staged_changes = subprocess.call(
+            [
+                "git", "diff", "--cached", "--quiet", "--exit-code"
+            ],
+            cwd=forge_file_directory
+        )
+        if has_staged_changes:
+            if commit:
+                git_args = [
+                    'git',
+                    'commit',
+                    '-m',
+                    'MNT: {}'.format(msg)
+                ]
+                if commit == "edit":
+                    git_args += [
+                        '--edit',
+                        '--status',
+                        '--verbose'
+                    ]
+                subprocess.check_call(
+                    git_args,
+                    cwd=forge_file_directory
+                )
+                print("")
+            else:
+                print(
+                    'You can commit the changes with:\n\n'
+                    '    git commit -m "MNT: {}"\n'.format(msg)
+                )
+            print("These changes need to be pushed to github!\n")
+        else:
+            print("No changes made. This feedstock is up-to-date.\n")
+
+
+def main(forge_file_directory, no_check_uptodate, commit):
+    error_on_warn = False if no_check_uptodate else True
+    index = conda_build.conda_interface.get_index(channel_urls=['conda-forge']) 
+    r = conda_build.conda_interface.Resolve(index)
+
+    # Check that conda-smithy is up-to-date
+    check_version_uptodate(r, "conda-smithy", __version__, error_on_warn)
+
     forge_dir = os.path.abspath(forge_file_directory)
-    config = _load_forge_config(forge_dir, variant_config_files)
+    print(os.listdir(os.path.join(forge_dir, "recipe")))
+    print(os.path.join(forge_dir, "recipe"))
+
+    config = _load_forge_config(forge_dir, None)
+    exclusive_config_file = config['exclusive_config_file']
+
+    if exclusive_config_file is None or not os.path.exists(exclusive_config_file):
```

Comment:
Now `conda-forge-pinning` is used as the `exclusive_config_file` if not given in `conda-forge.yml`. Config files inside the recipe is used by conda-build automatically.